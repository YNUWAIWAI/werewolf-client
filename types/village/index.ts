/* eslint no-unused-vars: 0, @typescript-eslint/no-unused-vars: 0 */
export * from './payload'

export const enum BaseContext {
  Base = 'https://werewolf.world/village/context/0.3/base.jsonld',
  Board = 'https://werewolf.world/village/context/0.3/board.jsonld',
  Character = 'https://werewolf.world/village/context/0.3/character.jsonld',
  Chat = 'https://werewolf.world/village/context/0.3/chat.jsonld',
  Error = 'https://werewolf.world/village/context/0.3/error.jsonld',
  FlavorText = 'https://werewolf.world/village/context/0.3/flavorText.jsonld',
  Role = 'https://werewolf.world/village/context/0.3/role.jsonld',
  Scroll = 'https://werewolf.world/village/context/0.3/scroll.jsonld',
  Star = 'https://werewolf.world/village/context/0.3/star.jsonld',
  Vote = 'https://werewolf.world/village/context/0.3/vote.jsonld',
  VotingResult = 'https://werewolf.world/village/context/0.3/votingResult.jsonld'
}
export const enum BoardState {
  CIRCLE = 'O',
  CROSS = 'X',
  FILL = 'fill',
  QUESTION = '?',
  TRIANGLE = 'Δ'
}
export const enum Channel {
  anonymousAudience = 'anonymousAudience',
  grave = 'grave',
  hunter = 'hunter',
  master = 'master',
  onymousAudience = 'onymousAudience',
  private = 'private',
  public = 'public',
  seer = 'seer',
  werewolf = 'werewolf'
}
export type CharacterId = string
export const enum CharacterStatus {
  alive = 'alive',
  dead = 'dead',
  deathByAttack = 'death by attack',
  deathByExecution = 'death by execution',
  deathByFear = 'death by fear',
  unnaturalDeath = 'unnatural death'
}
export const enum ChatChannel {
  grave = 'grave',
  master = 'master',
  private = 'private',
  public = 'public',
  werewolf = 'werewolf'
}
export type ChatId = string
export const enum ChatItemType {
  delimeter = 'delimeter',
  item = 'item'
}
export const enum Context {
  Avatar = 'https://werewolf.world/village/context/0.3/avatar.jsonld',
  Base = 'https://werewolf.world/village/context/0.3/base.jsonld',
  Board = 'https://werewolf.world/village/context/0.3/board.jsonld',
  BoardResult = 'https://werewolf.world/village/context/0.3/boardResult.jsonld',
  Character = 'https://werewolf.world/village/context/0.3/character.jsonld',
  Chat = 'https://werewolf.world/village/context/0.3/chat.jsonld',
  ChatSettings = 'https://werewolf.world/village/context/0.3/chatSettings.jsonld',
  Error = 'https://werewolf.world/village/context/0.3/error.jsonld',
  FlavorText = 'https://werewolf.world/village/context/0.3/flavorText.jsonld',
  Role = 'https://werewolf.world/village/context/0.3/role.jsonld',
  Scroll = 'https://werewolf.world/village/context/0.3/scroll.jsonld',
  Star = 'https://werewolf.world/village/context/0.3/star.jsonld',
  Village = 'https://werewolf.world/village/context/0.3/village.jsonld',
  Vote = 'https://werewolf.world/village/context/0.3/vote.jsonld',
  VotingResult = 'https://werewolf.world/village/context/0.3/votingResult.jsonld'
}
export const enum Directionality {
  serverToClient = 'server to client',
  clientToServer = 'client to server'
}
export const enum Gender {
  female = 'female',
  male = 'male'
}
export const enum InputChannel {
  grave = 'grave',
  postMortem = 'post mortem',
  private = 'private',
  public = 'public',
  werewolf = 'werewolf'
}
export const enum Language {
  en = 'en',
  fr = 'fr',
  it = 'it',
  ja = 'ja'
}
export interface LanguageMap {
  [Language.en]: string
  [Language.fr]?: string
  [Language.it]?: string
  [Language.ja]?: string
}
export type Payload$AgentId = number
export type Payload$ChatId = number
export const enum Phase {
  flavorText = 'flavor text',
  morning = 'morning',
  night = 'night',
  noon = 'noon',
  postMortem = 'post mortem',
  result = 'result'
}
export const enum Polarity {
  circle = 'O',
  cross = 'X',
  negative = 'negative',
  positive = 'positive',
  question = '?',
  triangle = 'Δ'
}
export const enum Result {
  lose = 'lose',
  win = 'win'
}
export const enum RoleId {
  hunter = 'hunter',
  madman = 'madman',
  mason = 'mason',
  master = 'master',
  medium = 'medium',
  seer = 'seer',
  villager = 'villager',
  werehamster = 'werehamster',
  werewolf = 'werewolf'
}
export const enum Severity {
  error = 'error',
  warning = 'warning'
}
export const enum SummaryType {
  audience = 'audience',
  character = 'character'
}
export const enum Team {
  villager = 'villager',
  werehamster = 'werehamster',
  werewolf = 'werewolf'
}
export type Token = string
export type VillageId = number
