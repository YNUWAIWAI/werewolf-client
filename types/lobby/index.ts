import {Avatar as AvatarScehma} from '../village/schema'
import {BuildVillage} from './schema'

export {
  Payload,
  PayloadType,
  Payload$AdvancedSearch,
  Payload$BuildVillage,
  Payload$Avatar,
  Payload$ChangeLanguage,
  Payload$ChangeUserEmail,
  Payload$ChangeUserName,
  Payload$ChangeUserPassword,
  Payload$EnterLobby,
  Payload$GetAvatar,
  Payload$GetSettings,
  Payload$IdSearch,
  Payload$KickOutPlayer,
  Payload$LeaveWaitingPage,
  Payload$Lobby,
  Payload$Ping,
  Payload$Play,
  Payload$Played,
  Payload$Pong,
  Payload$Ready,
  Payload$SearchResult,
  Payload$SelectVillage,
  Payload$Settings,
  Payload$WaitingPage
} from './payload'
export {Language} from '../village'
export const enum Avatar {
  fixed = 'fixed',
  random = 'random',
  unspecified = 'unspecified'
}
export const enum Member {
  A = 'A',
  B = 'B',
  C = 'C'
}
export const enum LobbyType {
  anonymousAudience = 'anonymous audience',
  onymousAudience = 'onymous audience',
  human = 'human player',
  robot = 'robot player'
}
export const enum PingStatus {
  danger = 'danger',
  safe = 'safe',
  warning = 'warning'
}
export const enum RoleId {
  hunter = 'hunter',
  madman = 'madman',
  mason = 'mason',
  medium = 'medium',
  seer = 'seer',
  villager = 'villager',
  werehamster = 'werehamster',
  werewolf = 'werewolf'
}
export type Village = Omit<BuildVillage, 'token' | 'type'>
export interface Cast {
  [numberOfPlayers: string]: {
    [member in Member]: BuildVillage['roleSetting']
  }
}
export type Token = AvatarScehma['token']
export interface WaitingPlayer {
  avatarImage: string
  isAnonymous: boolean
  isHost: boolean
  isMe: boolean
  name: string
  token: Token
}
export interface PingResult {
  ping: string
  status: PingStatus
  token: AvatarScehma['token']
}
export const enum TestStatus {
  notPassed = 'not passed',
  passed = 'passed'
}
export const enum Automation {
  full = 'full',
  semi = 'semi'
}
export const enum Authorized {
  no = 'no',
  yes = 'yes',
  waitForAcceptance = 'wait for acceptance',
}
export const enum AvatarStatus {
  awaitingAuthorization = 'awaiting authorization',
  awaitingCommunicationTest = 'awaiting communication test',
  connected = 'connected',
  runningInTheBackground = 'running in the background',
  runningInTheForeground = 'running in the foreground'
}
