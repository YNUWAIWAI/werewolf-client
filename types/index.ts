import * as lobby from './lobby'
import * as village from './village'

export {
  lobby,
  village
}
