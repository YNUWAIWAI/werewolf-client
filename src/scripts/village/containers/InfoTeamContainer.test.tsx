import * as React from 'react'
import {ImagePath} from '../constants/ImagePath'
import InfoTeamContainer from './InfoTeamContainer'
import IntlProviderContainer from './IntlProviderContainer'
import {Provider} from 'react-redux'
import {Role} from '../constants/Role'
import fakeStore from './fakeStore'
import {mount} from 'enzyme'
import {village} from '../types'

describe('<InfoTeamContainer />', () => {
  test('initialState', () => {
    const store = fakeStore()
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <InfoTeamContainer />
        </IntlProviderContainer>
      </Provider>
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
  test('mine.role exists', () => {
    const store = fakeStore(
      {
        mine: {
          role: {
            '@id': 'https://licos.online/state/0.2/village#3/role#seer',
            'id': village.RoleId.seer,
            'image': ImagePath.Role.seer,
            'name': Role.seer
          }
        }
      }
    )
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <InfoTeamContainer />
        </IntlProviderContainer>
      </Provider>
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
})
