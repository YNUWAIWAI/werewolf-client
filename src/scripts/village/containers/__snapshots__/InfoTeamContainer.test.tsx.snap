// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoTeamContainer /> initialState 1`] = `"<div class=\\"vi--info--team \\"><svg viewBox=\\"0 0 180 180\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M60,67c0-13 1-19 8-26c7-9 18-10 28-8c10,2 22,12 22,26c0,14-11,19-15,22c-7,2-10,6-11,11v20m0,12v16\\" fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"16\\"></path></svg></div>"`;

exports[`<InfoTeamContainer /> mine.role exists 1`] = `"<div class=\\"vi--info--team \\"><svg class=\\"vi--info--team--image teamVillager\\" viewBox=\\"0 0 50 50\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g><rect height=\\"3.01\\" width=\\"50\\"></rect><rect height=\\"3\\" width=\\"50\\" y=\\"47\\"></rect><rect height=\\"50\\" width=\\"3\\"></rect><rect height=\\"50\\" width=\\"3\\" x=\\"47\\"></rect></g><g><rect height=\\"3\\" transform=\\"translate(25 60.36) rotate(-135)\\" width=\\"35\\" x=\\"7.5\\" y=\\"23.5\\"></rect><rect height=\\"3\\" transform=\\"translate(-21.65 20.32) rotate(-45)\\" width=\\"35\\" x=\\"-3.8\\" y=\\"34.8\\"></rect><rect height=\\"3\\" transform=\\"translate(-1.43 15.44) rotate(-45)\\" width=\\"15\\" x=\\"10.43\\" y=\\"7.94\\"></rect><rect height=\\"3\\" transform=\\"translate(-10.8 38.07) rotate(-45)\\" width=\\"15\\" x=\\"33.06\\" y=\\"30.57\\"></rect><rect height=\\"3\\" transform=\\"translate(-4.47 22.79) rotate(-45)\\" width=\\"15\\" x=\\"17.77\\" y=\\"15.29\\"></rect><rect height=\\"3\\" transform=\\"translate(-7.68 30.53) rotate(-45)\\" width=\\"15\\" x=\\"25.52\\" y=\\"23.03\\"></rect></g></svg><span class=\\"vi--info--team--name\\">Villagers</span></div>"`;
